project/src/main/webapp/components/user-home/nav-panel.js

```diff
@@ -55,6 +60,28 @@ export class NavPanel extends LitElement {
     this.validDropdown = dropdown.value.length > 0;
   }
 
+  setPanelValue(e) {
+    this.value = e.target.value;
+    this.valueID = e.target.valueID;
+    this.createToggleFolderEvent();
+  }
+
+  setPanelValueAsMyDocs() {
+    this.value = '';
+    this.valueID = this.defaultFolderID;
+    this.createToggleFolderEvent();
+  }
+
+  createToggleFolderEvent() {
```

Comment:
Is it possible to combine this with createNewFolderEvent? i.e. a function like createEvent(string). This way you don't need to add a new function for each new event.